From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 14:50:11 -0600
Subject: (tmp-cherry-pick) Revert "Allow setting a DtlsTransportFactory in
 PeerConnectionDependencies" (4d0c4a2295)

This reverts commit d81865fb38bbdf7ba19f713338cbd8fdf25d513b.

Reason for revert: Breaks tests due to missing dep in api:peer_connection_interface

Original change's description:
> Allow setting a DtlsTransportFactory in PeerConnectionDependencies
>
> The JsepTransportController::Config already supports it, so this is
> just exposing up to PC clients.
>
> Note this required some work to avoid a circular dependency - changing
> to use forward declarations in dtls_transport_factory.h
>
> Bug: chromium:443019066
> Change-Id: I3b65fe36d88e25e40f54bafe858851aff723b16b
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/428281
> Commit-Queue: Tony Herre <herre@google.com>
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46317}

Bug: chromium:443019066
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: I144e0a51a752174c3bc0e33aec300cadea0b26dd
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/428760
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
Auto-Submit: Tony Herre <herre@google.com>
Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46318}
---
 api/BUILD.gn                      | 1 -
 api/DEPS                          | 1 -
 api/peer_connection_interface.h   | 2 --
 p2p/BUILD.gn                      | 2 ++
 p2p/dtls/dtls_transport_factory.h | 2 +-
 pc/BUILD.gn                       | 1 -
 pc/peer_connection.cc             | 2 --
 pc/peer_connection.h              | 2 --
 8 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/api/BUILD.gn b/api/BUILD.gn
index c0af7ec4b7..c3227ccfc7 100644
--- a/api/BUILD.gn
+++ b/api/BUILD.gn
@@ -582,7 +582,6 @@ rtc_library("libjingle_peerconnection_api") {
     "../call:rtp_interfaces",
     "../media:media_engine",
     "../p2p:connection",
-    "../p2p:dtls_transport_factory",
     "../p2p:port",
     "../p2p:port_allocator",
     "../pc:media_factory",
diff --git a/api/DEPS b/api/DEPS
index 490db60efb..2460690e4a 100644
--- a/api/DEPS
+++ b/api/DEPS
@@ -147,7 +147,6 @@ specific_include_rules = {
     "+media/base/media_engine.h",
     "+p2p/base/port.h",
     "+p2p/base/port_allocator.h",
-    "+p2p/dtls/dtls_transport_factory.h",
     "+rtc_base/network.h",
     "+rtc_base/network_constants.h",
     "+rtc_base/network_monitor_factory.h",
diff --git a/api/peer_connection_interface.h b/api/peer_connection_interface.h
index 9ccc8c62b8..5e5632ce7f 100644
--- a/api/peer_connection_interface.h
+++ b/api/peer_connection_interface.h
@@ -134,7 +134,6 @@
 #include "api/units/time_delta.h"
 #include "p2p/base/port.h"
 #include "p2p/base/port_allocator.h"
-#include "p2p/dtls/dtls_transport_factory.h"
 #include "rtc_base/checks.h"
 #include "rtc_base/network.h"
 #include "rtc_base/network_constants.h"
@@ -1404,7 +1403,6 @@ struct RTC_EXPORT PeerConnectionDependencies final {
   // Factory for creating resolvers that look up hostnames in DNS
   std::unique_ptr<AsyncDnsResolverFactoryInterface> async_dns_resolver_factory;
   std::unique_ptr<IceTransportFactory> ice_transport_factory;
-  std::unique_ptr<DtlsTransportFactory> dtls_transport_factory;
   std::unique_ptr<RTCCertificateGeneratorInterface> cert_generator;
   std::unique_ptr<SSLCertificateVerifier> tls_cert_verifier;
   std::unique_ptr<VideoBitrateAllocatorFactory> video_bitrate_allocator_factory;
diff --git a/p2p/BUILD.gn b/p2p/BUILD.gn
index 3b0b2efea8..e4e7e43f23 100644
--- a/p2p/BUILD.gn
+++ b/p2p/BUILD.gn
@@ -333,6 +333,8 @@ rtc_library("dtls_transport") {
 rtc_source_set("dtls_transport_factory") {
   sources = [ "dtls/dtls_transport_factory.h" ]
   deps = [
+    ":dtls_transport_internal",
+    ":ice_transport_internal",
     "../api:ice_transport_interface",
     "../api:scoped_refptr",
     "../api/crypto:options",
diff --git a/p2p/dtls/dtls_transport_factory.h b/p2p/dtls/dtls_transport_factory.h
index 2888798175..f0d7d971ab 100644
--- a/p2p/dtls/dtls_transport_factory.h
+++ b/p2p/dtls/dtls_transport_factory.h
@@ -16,10 +16,10 @@
 #include "api/crypto/crypto_options.h"
 #include "api/ice_transport_interface.h"
 #include "api/scoped_refptr.h"
+#include "p2p/dtls/dtls_transport_internal.h"
 #include "rtc_base/ssl_stream_adapter.h"
 
 namespace webrtc {
-class DtlsTransportInternal;
 
 // This interface is used to create DTLS transports. The external transports
 // can be injected into the JsepTransportController through it.
diff --git a/pc/BUILD.gn b/pc/BUILD.gn
index 78fdbfe019..d032de20fe 100644
--- a/pc/BUILD.gn
+++ b/pc/BUILD.gn
@@ -1340,7 +1340,6 @@ rtc_library("peer_connection") {
     "../media:stream_params",
     "../modules/rtp_rtcp:rtp_rtcp_format",
     "../p2p:connection_info",
-    "../p2p:dtls_transport_factory",
     "../p2p:dtls_transport_internal",
     "../p2p:ice_transport_internal",
     "../p2p:p2p_constants",
diff --git a/pc/peer_connection.cc b/pc/peer_connection.cc
index 0b9e605e2d..c169e6c18e 100644
--- a/pc/peer_connection.cc
+++ b/pc/peer_connection.cc
@@ -598,7 +598,6 @@ PeerConnection::PeerConnection(
       port_allocator_(std::move(dependencies.allocator)),
       lna_permission_factory_(std::move(dependencies.lna_permission_factory)),
       ice_transport_factory_(std::move(dependencies.ice_transport_factory)),
-      dtls_transport_factory_(std::move(dependencies.dtls_transport_factory)),
       tls_cert_verifier_(std::move(dependencies.tls_cert_verifier)),
       call_(std::move(call)),
       network_thread_safety_(
@@ -752,7 +751,6 @@ JsepTransportController* PeerConnection::InitializeNetworkThread(
   }
 
   config.ice_transport_factory = ice_transport_factory_.get();
-  config.dtls_transport_factory = dtls_transport_factory_.get();
   config.on_dtls_handshake_error =
       [weak_ptr = weak_factory_.GetWeakPtr()](SSLHandshakeError s) {
         if (weak_ptr) {
diff --git a/pc/peer_connection.h b/pc/peer_connection.h
index 766243f6bf..e8b6d52799 100644
--- a/pc/peer_connection.h
+++ b/pc/peer_connection.h
@@ -65,7 +65,6 @@
 #include "p2p/base/port.h"
 #include "p2p/base/port_allocator.h"
 #include "p2p/base/transport_description.h"
-#include "p2p/dtls/dtls_transport_factory.h"
 #include "pc/channel_interface.h"
 #include "pc/codec_vendor.h"
 #include "pc/connection_context.h"
@@ -677,7 +676,6 @@ class PeerConnection : public PeerConnectionInternal,
                                // pointer is given to
                                // `jsep_transport_controller_` and used on the
                                // network thread.
-  const std::unique_ptr<DtlsTransportFactory> dtls_transport_factory_;
   const std::unique_ptr<SSLCertificateVerifier> tls_cert_verifier_
       RTC_GUARDED_BY(network_thread());
 
